syntax="proto3";
package proto;



/**
currency type
 */
enum CostType{
  Gems=0;       //use gems
  Gold=1;       //use coin
}

/**
client give a building upgrade request
 */
message ReqUpgradeBuilding{
  string playerId = 1;
  int64 buildingId = 2;
  CostType costBy = 3;
}

/**
return upgrade result
 */
message RspUpgradeBuilding{
  int64 buildingId=1;         //upgraded buildingId
  int32 buildingLevelNow=2;   //upgraded building Level
  int64 playerGemNow=3;       //player gems num
  int64 playerGoldNow=4;      //player gold num
  bool  isSuccess=5;           //is success
  string errorMsg=6;          //error message
}